src/Microsoft.AspNet.FileProviders/Implementation/PhysicalFilesWatcher.cs

```diff
@@ -148,10 +148,10 @@ private string NormalizeFilter(string filter)
 
         private bool IsDirectoryPath(string path)
         {
-            return path != null && path.Length > 1 &&
-                (path[path.Length - 1] == Path.DirectorySeparatorChar ||
-                path[path.Length - 1] == Path.AltDirectorySeparatorChar);
-        }
+			if (path == null)
+				throw new ArgumentNullException();
```

Comment:
Hi @hishamco. The point is not about how the `ArgumentNullException` be thrown but about whether it should be thrown. You see previously the code return `false` when the `path` is null. The client codes may rely on this behavior. Now if the contract changed to thrown exception, then it's a behavior change and it's a breaking change.